echopype/convert/set_groups_base.py

```diff
@@ -396,18 +396,28 @@ def _add_index_data_to_platform_ds(
         )
         platform_ds = platform_ds.assign(
             {
+                "ping_number": xr.DataArray(
+                    np.array(self.parser_obj.idx["ping_number"]),
+                    dims=("time3"),
+                    coords={"time3": timestamp_array},
+                ),
+                "file_offset": xr.DataArray(
+                    np.array(self.parser_obj.idx["file_offset"]),
+                    dims=("time3"),
+                    coords={"time3": timestamp_array},
+                ),
                 "vessel_distance": xr.DataArray(
                     np.array(self.parser_obj.idx["vessel_distance"]),
                     dims=("time3"),
                     coords={"time3": timestamp_array},
                 ),
```

Comment:
Just to make it absolutely clear where these came from
```suggestion
                "idx_ping_number": xr.DataArray(
                    np.array(self.parser_obj.idx["ping_number"]),
                    dims=("time3"),
                    coords={"time3": timestamp_array},
                ),
                "idx_file_offset": xr.DataArray(
                    np.array(self.parser_obj.idx["file_offset"]),
                    dims=("time3"),
                    coords={"time3": timestamp_array},
                ),
                "idx_vessel_distance": xr.DataArray(
                    np.array(self.parser_obj.idx["vessel_distance"]),
                    dims=("time3"),
                    coords={"time3": timestamp_array},
                ),
```

Reply:
And just because I am looking at this... why wouldn't you do the attribute assignment here in each of the xr.DataArray too?

Reply:
Ah yeah you're right I'll do that